diff --git a/as31/run.c b/as31/run.c
index 9e5263b..6010ee8 100644
--- a/as31/run.c
+++ b/as31/run.c
@@ -34,10 +34,10 @@ int run_as31(const char *infile, int lst, int use_stdout,
 	FILE* finPre;
 	char tmpName[256];
 	char *lineBuffer=NULL;
-	int sizeBuf=0;
+	long unsigned int sizeBuf=0;
 	char *includePtr=NULL;
 	char *incLineBuffer=NULL;
-	int incSizeBuf=0;
+	long unsigned int incSizeBuf=0;
 	FILE* includeFile=NULL;
 	int fd;
 
